### PR TITLE
[move-lang] Only stop at blocking errors

### PR DESCRIPTION
**Pull request by tnowacki**
- Used the new severity system to stop only at blocking errors
- Fixed a few things in hlir/translate that needed to be reworked to deal with ill-typed programs

## Motivation

- Trying to have to stop compilation as little as possible. Especially with warnings
- Still need to fix things around the parser at some point

## Test Plan

- Updated tests

### CODE REVIEW

File: language/move-lang/src/hlir/translate.rs [hunk 1]
```diff
@@ -1168,16 +1194,30 @@ fn exp_impl(
                     .map(|(e, (f, bt))| (f, bt, e))
                     .collect()
             } else {
-                let mut fields = (0..decl_fields.len()).map(|_| None).collect::<Vec<_>>();
+                let num_fields = decl_fields.as_ref().map(|m| m.len()).unwrap_or(0);
+                let mut fields = (0..num_fields).map(|_| None).collect::<Vec<_>>();
                 for (decl_idx, f, _exp_idx, bt, tf) in texp_fields {
+                    // Might have too many arguments, there will be an error from typing
```
Comment (tzakian) [inline language/move-lang/src/hlir/translate.rs:1200]: This comment is making me wonder if eventually we should have a way of specifying  "nonblocking error X will result in blocking errors A, B, or C" later on in compilation.

**Review by tzakian — APPROVED** [review]
LGTM! This is awesome.

File: language/move-lang/src/hlir/translate.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/hlir/translate.rs:1200]: Yeah... I've thought about something like this but I think they would be very hard to maintain. Maybe would be good to check that there is a given error in the diags though

**Review by tzakian — APPROVED** [review]
New changes look good to me. I'm thinking it might be good to add a `-Werror`-type flag to the compiler at some point soon now that we have warnings.

**Pull request merged**